linkerd/app/src/env.rs

```diff
@@ -793,29 +812,58 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
         .unwrap_or(super::tap::Config::Disabled);
 
     let identity = {
-        let (addr, certify, tls) = identity_config?;
-        // If the address doesn't have a server identity, then we're on localhost.
-        let connect = if addr.addr.is_loopback() {
-            inbound.proxy.connect.clone()
-        } else {
-            outbound.proxy.connect.clone()
-        };
-        let failfast_timeout = if addr.addr.is_loopback() {
-            inbound.http_request_queue.failfast_timeout
-        } else {
-            outbound.http_request_queue.failfast_timeout
-        };
-        identity::Config::Linkerd {
-            certify,
-            tls,
-            client: ControlConfig {
-                addr,
-                connect,
-                buffer: QueueConfig {
-                    capacity: DEFAULT_CONTROL_QUEUE_CAPACITY,
-                    failfast_timeout,
-                },
+        let tls = tls?;
+
+        match strings.get(ENV_IDENTITY_SPIRE_SOCKET)? {
+            Some(socket) => match &tls.id {
+                // TODO: perform stricter SPIFFE ID validation following:
+                // https://github.com/spiffe/spiffe/blob/27b59b81ba8c56885ac5d4be73b35b9b3305fd7a/standards/SPIFFE-ID.md
+                identity::Id::Uri(uri) if uri.scheme() == SPIFFE_ID_URI_SCHEME => {
```

Comment:
i'm not sure about the semantics of SPIFFE but you may want to use [`eq_ignore_ascii_case`](https://doc.rust-lang.org/std/string/struct.String.html#method.eq_ignore_ascii_case)